crates/anstyle-svg/src/lib.rs

```diff
@@ -306,6 +273,45 @@ impl Term {
         writeln!(&mut buffer, r#"</html>"#).unwrap();
         buffer
     }
+
+    fn render_classes(&self, buffer: &mut String, elements: &[adapter::Element]) {
+        use std::fmt::Write as _;
+
+        let fg_color = rgb_value(self.fg_color, self.palette);
+        let bg_color = rgb_value(self.bg_color, self.palette);
+        let font_family = self.font_family;
+
+        writeln!(buffer, r#"    .{FG} {{ color: {fg_color} }}"#).unwrap();
+        writeln!(buffer, r#"    .{BG} {{ background: {bg_color} }}"#).unwrap();
+        for (name, rgb) in color_styles(elements, self.palette) {
+            if name.starts_with(FG_PREFIX) {
+                writeln!(buffer, r#"    .{name} {{ color: {rgb} }}"#).unwrap();
+            }
+            if name.starts_with(BG_PREFIX) {
+                writeln!(
+                    buffer,
+                    r#"    .{name} {{ background: {rgb}; user-select: none; }}"#
+                )
+                .unwrap();
+            }
+            if name.starts_with(UNDERLINE_PREFIX) {
+                writeln!(
+                    buffer,
+                    r#"    .{name} {{ text-decoration-line: underline; text-decoration-color: {rgb} }}"#
+                )
+                .unwrap();
+            }
+        }
+        writeln!(buffer, r#"    .container {{"#).unwrap();
+        writeln!(buffer, r#"      line-height: {LINE_HEIGHT}px;"#).unwrap();
+        writeln!(buffer, r#"    }}"#).unwrap();
+        write_effects_in_use(buffer, elements);
+        writeln!(buffer, r#"    span {{"#).unwrap();
+        writeln!(buffer, r#"      font: 14px {font_family};"#).unwrap();
+        writeln!(buffer, r#"      white-space: pre;"#).unwrap();
+        writeln!(buffer, r#"      line-height: {LINE_HEIGHT}px;"#).unwrap();
+        writeln!(buffer, r#"    }}"#).unwrap();
```

Comment:
Do we need to do anything to make the classes more unique if this is intended to be mixed in with other HTML?

Reply:
For now I don't need it but providing a way for users to associate a color with a class name would be nice. Something like:

```rust
struct ColorOptions {
    bg: Colors,
    fg: Colors,
}

struct Colors {
    red: Option<String>,
    blue: Option<String>,
    // ...
}
```

If `None`, then we use the default name, etc.